### ISSUE TITLE
dbcontext optimize command omits certain characters from namespaces

### ISSUE THREAD

**Issue by chriscameron-vertexinc** [dotnet/efcore#30045]
EF Core version: 7.0.1
Database provider: Npgsql.EntityFrameworkCore.PostgreSQL
Target framework: .NET 7.0
Operating system: Windows 10
IDE: Visual Studio 2022 17.4.2

I am exploring the use of Compiled Models in my Entity Framework projects to reduce warmup time.

I'm running the command:
`dotnet ef dbcontext optimize -o Model/Optimized -n 'My_Namespace' -c MyContext -v`

And ending up with generated code in the namespace:
`MyNamespace`

```
// <auto-generated />
using System;
using System.Reflection;
using Microsoft.EntityFrameworkCore.Metadata;

#pragma warning disable 219, 612, 618
#nullable enable

namespace MyNamespace
{
```

I thought it may have been an escaping issue, so I tried:
`dotnet ef dbcontext optimize -o Model/Optimized -n 'My\_Namespace' -c MyContext -v`

And still ended up with:
`MyNamespace`

The output from the command ends with the following, incorrect instruction:
`Successfully generated a compiled model, to use it call 'options.UseModel(My\_Namespace.MyContextModel.Instance)'. Run this command again when the model is modified.`

**Comment by ErikEJ** [comment]
I think this will be fixed in 7.0.3

**Comment by ajcvickers** [comment]
Likely duplicate of #29450.
